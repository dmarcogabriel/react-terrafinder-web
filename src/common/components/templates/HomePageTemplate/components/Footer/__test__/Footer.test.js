import React from 'react';
import ReactDOM from 'react-dom';
import renderer from 'react-test-renderer';
import { render, fireEvent } from '@testing-library/react';
import { Footer } from '..';

jest.mock('hooks/useNotification', () => ({
  useNotification: () => ({
    showNotification: jest.fn(),
  }),
}));

let mockUrl;
const mockPush = jest.fn((url) => {
  mockUrl = url;
});
jest.mock('react-router-dom', () => ({
  BrowserRouter: ({ children }) => <div>{children}</div>,
  useHistory: () => ({
    push: mockPush,
  }),
}));

beforeEach(() => {
  mockUrl = null;
});

describe('<Footer />', () => {
  it('renders without crashing', () => {
    const div = document.createElement('div');
    ReactDOM.render(<Footer />, div);
    ReactDOM.unmountComponentAtNode(div);
  });

  it('matches snapshot', () => {
    const tree = renderer.create(<Footer />).toJSON();
    expect(tree).toMatchSnapshot();
  });

  describe('footer navigation', () => {
    it('should navigate to home', () => {
      const { getByTestId } = render(<Footer />);

      const listItem = getByTestId('link-t1-l1');
      fireEvent.click(listItem);

      expect(mockPush).toHaveBeenCalled();
      expect(mockUrl).toBe('/');
    });

    it('should navigate to about', () => {
      const { getByTestId } = render(<Footer />);

      const listItem = getByTestId('link-t1-l2');
      fireEvent.click(listItem);

      expect(mockUrl).toBe('/about');
    });

    it('should navigate to privacy policy', () => {
      const { getByTestId } = render(<Footer />);

      const listItem = getByTestId('link-t1-l3');
      fireEvent.click(listItem);

      expect(mockUrl).toBe('/privacy-policy');
    });

    it('should navigate to support', () => {
      const { getByTestId } = render(<Footer />);

      const listItem = getByTestId('link-t1-l4');
      fireEvent.click(listItem);

      expect(mockUrl).toBe('/support');
    });

    it('should navigate to create property', () => {
      const { getByTestId } = render(<Footer />);

      const listItem = getByTestId('link-t2-l1');
      fireEvent.click(listItem);

      expect(mockUrl).toBe('/create-property');
    });

    it('should navigate to search property', () => {
      const { getByTestId } = render(<Footer />);

      const listItem = getByTestId('link-t2-l2');
      fireEvent.click(listItem);

      expect(mockUrl).toBe('/search-property');
    });

    it('should navigate to prices and plans', () => {
      const { getByTestId } = render(<Footer />);

      const listItem = getByTestId('link-t2-l3');
      fireEvent.click(listItem);

      expect(mockUrl).toBe('/plans');
    });
  });
});
