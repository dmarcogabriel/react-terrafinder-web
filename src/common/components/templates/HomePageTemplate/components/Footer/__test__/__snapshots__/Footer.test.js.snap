// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Footer /> matches snapshot 1`] = `
<footer
  className="MuiBox-root css-108tvhz"
>
  <hr
    className="MuiDivider-root MuiDivider-fullWidth css-tz2ti8-MuiDivider-root"
  />
  <div
    className="MuiBox-root css-zitybv"
  >
    <h3
      className="MuiTypography-root MuiTypography-h3 css-15anx50-MuiTypography-root"
    >
      Terrafinder
    </h3>
    <ul
      className="MuiList-root MuiList-padding css-12n6fjr-MuiList-root"
    >
      <h6
        className="MuiTypography-root MuiTypography-h6 css-xqtpgv-MuiTypography-root"
      >
        Institucional
      </h6>
      <li
        className="MuiListItem-root MuiListItem-gutters css-1nxmd3h-MuiListItem-root"
        disabled={false}
      >
        <div
          className="MuiListItemButton-root MuiListItemButton-gutters MuiButtonBase-root css-cvhtoe-MuiButtonBase-root-MuiListItemButton-root"
          data-testid="link-t1-l1"
          onBlur={[Function]}
          onClick={[Function]}
          onContextMenu={[Function]}
          onDragLeave={[Function]}
          onFocus={[Function]}
          onKeyDown={[Function]}
          onKeyUp={[Function]}
          onMouseDown={[Function]}
          onMouseLeave={[Function]}
          onMouseUp={[Function]}
          onTouchEnd={[Function]}
          onTouchMove={[Function]}
          onTouchStart={[Function]}
          role="button"
          tabIndex={0}
        >
          Página Inicial
        </div>
      </li>
      <li
        className="MuiListItem-root MuiListItem-gutters css-1nxmd3h-MuiListItem-root"
        disabled={false}
      >
        <div
          className="MuiListItemButton-root MuiListItemButton-gutters MuiButtonBase-root css-cvhtoe-MuiButtonBase-root-MuiListItemButton-root"
          data-testid="link-t1-l2"
          onBlur={[Function]}
          onClick={[Function]}
          onContextMenu={[Function]}
          onDragLeave={[Function]}
          onFocus={[Function]}
          onKeyDown={[Function]}
          onKeyUp={[Function]}
          onMouseDown={[Function]}
          onMouseLeave={[Function]}
          onMouseUp={[Function]}
          onTouchEnd={[Function]}
          onTouchMove={[Function]}
          onTouchStart={[Function]}
          role="button"
          tabIndex={0}
        >
          Quem somos
        </div>
      </li>
      <li
        className="MuiListItem-root MuiListItem-gutters css-1nxmd3h-MuiListItem-root"
        disabled={false}
      >
        <div
          className="MuiListItemButton-root MuiListItemButton-gutters MuiButtonBase-root css-cvhtoe-MuiButtonBase-root-MuiListItemButton-root"
          data-testid="link-t1-l3"
          onBlur={[Function]}
          onClick={[Function]}
          onContextMenu={[Function]}
          onDragLeave={[Function]}
          onFocus={[Function]}
          onKeyDown={[Function]}
          onKeyUp={[Function]}
          onMouseDown={[Function]}
          onMouseLeave={[Function]}
          onMouseUp={[Function]}
          onTouchEnd={[Function]}
          onTouchMove={[Function]}
          onTouchStart={[Function]}
          role="button"
          tabIndex={0}
        >
          Política de privacidade
        </div>
      </li>
      <li
        className="MuiListItem-root MuiListItem-gutters css-1nxmd3h-MuiListItem-root"
        disabled={false}
      >
        <div
          className="MuiListItemButton-root MuiListItemButton-gutters MuiButtonBase-root css-cvhtoe-MuiButtonBase-root-MuiListItemButton-root"
          data-testid="link-t1-l4"
          onBlur={[Function]}
          onClick={[Function]}
          onContextMenu={[Function]}
          onDragLeave={[Function]}
          onFocus={[Function]}
          onKeyDown={[Function]}
          onKeyUp={[Function]}
          onMouseDown={[Function]}
          onMouseLeave={[Function]}
          onMouseUp={[Function]}
          onTouchEnd={[Function]}
          onTouchMove={[Function]}
          onTouchStart={[Function]}
          role="button"
          tabIndex={0}
        >
          Fale Conosco
        </div>
      </li>
    </ul>
    <ul
      className="MuiList-root MuiList-padding css-12n6fjr-MuiList-root"
    >
      <h6
        className="MuiTypography-root MuiTypography-h6 css-xqtpgv-MuiTypography-root"
      >
        Funcionalidades
      </h6>
      <li
        className="MuiListItem-root MuiListItem-gutters css-1nxmd3h-MuiListItem-root"
        disabled={false}
      >
        <div
          className="MuiListItemButton-root MuiListItemButton-gutters MuiButtonBase-root css-cvhtoe-MuiButtonBase-root-MuiListItemButton-root"
          data-testid="link-t2-l1"
          onBlur={[Function]}
          onClick={[Function]}
          onContextMenu={[Function]}
          onDragLeave={[Function]}
          onFocus={[Function]}
          onKeyDown={[Function]}
          onKeyUp={[Function]}
          onMouseDown={[Function]}
          onMouseLeave={[Function]}
          onMouseUp={[Function]}
          onTouchEnd={[Function]}
          onTouchMove={[Function]}
          onTouchStart={[Function]}
          role="button"
          tabIndex={0}
        >
          Quero anunciar meu imóvel
        </div>
      </li>
      <li
        className="MuiListItem-root MuiListItem-gutters css-1nxmd3h-MuiListItem-root"
        disabled={false}
      >
        <div
          className="MuiListItemButton-root MuiListItemButton-gutters MuiButtonBase-root css-cvhtoe-MuiButtonBase-root-MuiListItemButton-root"
          data-testid="link-t2-l2"
          onBlur={[Function]}
          onClick={[Function]}
          onContextMenu={[Function]}
          onDragLeave={[Function]}
          onFocus={[Function]}
          onKeyDown={[Function]}
          onKeyUp={[Function]}
          onMouseDown={[Function]}
          onMouseLeave={[Function]}
          onMouseUp={[Function]}
          onTouchEnd={[Function]}
          onTouchMove={[Function]}
          onTouchStart={[Function]}
          role="button"
          tabIndex={0}
        >
          Quero comprar um imóvel
        </div>
      </li>
      <li
        className="MuiListItem-root MuiListItem-gutters css-1nxmd3h-MuiListItem-root"
        disabled={false}
      >
        <div
          className="MuiListItemButton-root MuiListItemButton-gutters MuiButtonBase-root css-cvhtoe-MuiButtonBase-root-MuiListItemButton-root"
          data-testid="link-t2-l3"
          onBlur={[Function]}
          onClick={[Function]}
          onContextMenu={[Function]}
          onDragLeave={[Function]}
          onFocus={[Function]}
          onKeyDown={[Function]}
          onKeyUp={[Function]}
          onMouseDown={[Function]}
          onMouseLeave={[Function]}
          onMouseUp={[Function]}
          onTouchEnd={[Function]}
          onTouchMove={[Function]}
          onTouchStart={[Function]}
          role="button"
          tabIndex={0}
        >
          Preços e planos
        </div>
      </li>
    </ul>
    <h6
      className="MuiTypography-root MuiTypography-h6 css-xqtpgv-MuiTypography-root"
    >
      Fique por dentro das novidades!
    </h6>
    <p
      className="MuiTypography-root MuiTypography-body1 css-15wu8wo-MuiTypography-root"
    >
      Se inscreva em nossa newsletter para receber os anúncios com os melhores preços!
    </p>
    <form
      className="MuiPaper-root MuiPaper-elevation MuiPaper-rounded MuiPaper-elevation1 css-np4jd-MuiPaper-root"
      noValidate={true}
      onSubmit={[Function]}
    >
      <div
        className="MuiInputBase-root MuiInputBase-colorPrimary MuiInputBase-fullWidth css-13bnupw-MuiInputBase-root"
        onClick={[Function]}
      >
        <input
          aria-invalid={false}
          className="MuiInputBase-input css-yz9k0d-MuiInputBase-input"
          data-testid="emailInput"
          inputMode="email"
          onAnimationStart={[Function]}
          onBlur={[Function]}
          onChange={[Function]}
          onFocus={[Function]}
          placeholder="Informe seu e-mail aqui"
          type="text"
          value=""
        />
      </div>
      <button
        className="MuiButton-root MuiButton-contained MuiButton-containedSuccess MuiButton-sizeMedium MuiButton-containedSizeMedium MuiButtonBase-root css-1h0hyhf-MuiButtonBase-root-MuiButton-root"
        data-testid="registerEmail"
        disabled={false}
        onBlur={[Function]}
        onClick={[Function]}
        onContextMenu={[Function]}
        onDragLeave={[Function]}
        onFocus={[Function]}
        onKeyDown={[Function]}
        onKeyUp={[Function]}
        onMouseDown={[Function]}
        onMouseLeave={[Function]}
        onMouseUp={[Function]}
        onTouchEnd={[Function]}
        onTouchMove={[Function]}
        onTouchStart={[Function]}
        tabIndex={0}
        type="button"
      >
        Cadastrar
      </button>
    </form>
  </div>
  <hr
    className="MuiDivider-root MuiDivider-fullWidth css-tz2ti8-MuiDivider-root"
  />
  <h6
    className="MuiTypography-root MuiTypography-subtitle2 css-qwq8wp-MuiTypography-root"
  >
    Copyright C 2020 Terrafinder. Todos os direitos reservados.
  </h6>
</footer>
`;
